cli/core/mpt/flows.py

```
@@ -27,21 +24,21 @@ def get_products(
     """
-    url = f"/catalog/products?limit={quote_plus(str(limit))}&offset={quote_plus(str(offset))}"
+    product_collection = mpt_client.catalog.products
     if query:
-        url = f"{url}&{quote_plus(query)}"
-    response = mpt_client.get(url)
-    response.raise_for_status()
-    json_body = response.json()
+        product_collection = product_collection.filter(RQLQuery.from_string(query))
+    product_response = product_collection.fetch_page(limit=limit, offset=offset)
+    if product_response.meta is None:
+        raise MPTAPIError("Missing pagination metadata in product response.", "Invalid response")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)